src/Drivers/Zibal/Zibal.php

```diff
@@ -84,13 +84,21 @@ public function purchase()
             $description = $this->settings->description;
         }
 
+        $allowedCards = null;
+        if (!empty($details['allowedCards'])) {
+            $allowedCards = $details['allowedCards'];
+        } else {
+            $allowedCards = $this->settings->allowedCards;
```

Comment:
where this setting comes from? i mean `allowedCards`
what would happen if we go to the `else` block and `allowedCards` not exists?

Reply:
You are right. I added allowedCards to the config/payment file and didnt check for null. Pushed new commit which fixes that issue. Now it checks both user input and settings file.